myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -105,11 +106,20 @@ public void onEvent(ResourceOffersEvent event, long sequence,
         for (NodeTask nodeTask : nodeTasks) {
           nodeTask.setSlaveAttributes(offer.getAttributesList());
         }
+        // keep this in case SchedulerState gets out of sync. This should not happen with 
+        // synchronizing addNodes method in SchedulerState
+        // but to keep it safe
+        final Set<Protos.TaskID> blacklistedTasks = Sets.newHashSet();
```

Comment:
I see it now. It's removing from pendingTasks as well. Makes sense to do it the way you are doing right now. 
However, we should give the collection a better name. ```blacklistedTasks``` can be misleading since we are not really black listing anything. How about ```missingTasks``` ?